.github/workflows/nightly-release.yml

```diff
@@ -0,0 +1,111 @@
+# This GitHub action triggers a fresh set of Nomad builds
+# and publishes them to GitHub Releases under the `nightly` tag.
+# Note that artifacts available via GitHub Releases are not codesigned or notarized.
+# Failures are reported to slack.
+#
+
+name: Nightly Release
+
+on:
+  # TODO: remove the push event before merging this PR; this is for testing purposes
+  schedule:
+    # Runs against the default branch every day at midnight
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  # Build a fresh set of artifacts
+  build-artifacts:
+    # TODO: change branch from add-nightlies to main before merging this PR
+    uses: hashicorp/nomad/.github/workflows/build.yml@main
+  github-release:
+    needs: build-artifacts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download built artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: out/
+      # Set BUILD_OUTPUT_LIST to out\<project>-<version>.<fileext>\*,out\...
+      # This is needed to attach the build artifacts to the GitHub Release
+      - name: Set BUILD_OUTPUT_LIST
+        run: |
+          echo "$(ls -xm1 out/)" > tmp.txt
+          cat tmp.txt | sed 's:.*:out/&/*:' > tmp2.txt
+          echo "BUILD_OUTPUT_LIST=$(cat tmp2.txt | tr '\n' ',' | perl -ple 'chop')" >> $GITHUB_ENV
+          rm -rf tmp.txt && rm -rf tmp2.txt
+      - name: Advance nightly tag
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            try {
+                await github.git.deleteRef({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  ref: "tags/nightly"
+                })
+            } catch (e) {
+              console.log("Warning: The nightly tag doesn't exist yet, so there's nothing to do. Trace: " + e)
+            }
+            await github.git.createRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: "refs/tags/nightly",
+              sha: context.sha
+            })
+      # This will create a new GitHub Release called `nightly`
+      # If a release with this name already exists, it will overwrite the existing data
+      - name: Create a nightly GitHub prerelease
+        id: create_prerelease
+        uses: ncipollo/release-action@v1
+        with:
+          name: nightly
+          artifacts: "${{ env.BUILD_OUTPUT_LIST }}"
+          tag: nightly
+          bodyFile: ".github/workflows/nightly-release-readme.md"
+          prerelease: true
+          allowUpdates: true
+          removeArtifacts: true
+          draft: false
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Publish nightly GitHub prerelease
+        uses: eregon/publish-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          release_id: ${{ steps.create_prerelease.outputs.id }}
```

Comment:
Out of curiosity, what does this action do? I read their docs but still could quite get what "publish the release" means in this case.

Reply:
I believe this action takes the `nightly` Github prerelease (previous action) and communicates with Github and automates the process of pushing a nightly release + tag for users to utilize. cc: @sarahethompson to correct/clarify me if I'm wrong!